core/src/main/java/com/linecorp/armeria/client/ClientTlsSpec.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import javax.net.ssl.KeyManagerFactory;
+
+import com.linecorp.armeria.common.AbstractTlsSpec;
+import com.linecorp.armeria.common.TlsKeyPair;
+import com.linecorp.armeria.common.TlsPeerVerifierFactory;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.util.TlsEngineType;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+/**
+ * Specifies TLS configuration for client connections.
+ */
+@UnstableApi
+public final class ClientTlsSpec extends AbstractTlsSpec {
+
+    private static final ClientTlsSpec DEFAULT = builder().build();
+
+    ClientTlsSpec(Set<String> protocols, Set<String> alpnProtocols, Set<String> ciphers,
```

Comment:
Shouldn't it be `tlsVersions` instead of `protocols`?
Also, how about validating the values?

Reply:
Thanks for pointing that out. 😓  I wasn't aware of that convention. My usual preference is to align with Netty's naming for consistency, but I felt a different name was more descriptive in this specific context.
But it's your call, of course. I'm fine either way. 😉